hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -970,10 +979,11 @@ public void runMetaSync() {
         } catch (HoodieException e) {
           LOG.info("SyncTool class " + impl.trim() + " failed with exception", e);
           metaSyncExceptions.add(e);
+          implsFailed.add(impl.trim());
         }
       }
       if (!metaSyncExceptions.isEmpty()) {
-        throw SyncUtilHelpers.getExceptionFromList(metaSyncExceptions);
+        throw new HoodieDeltaStreamerMetaSyncException("Meta sync failure for " + String.join(",", implsFailed), SyncUtilHelpers.getExceptionFromList(metaSyncExceptions));
```

Comment:
It may be easier to work with the exception if we have it take in the list of `implsfailed` and then construct the message in the call to `super`